lms/djangoapps/teams/views.py

```diff
@@ -368,6 +368,9 @@ def post(self, request):
             }, status=status.HTTP_400_BAD_REQUEST)
         else:
             team = serializer.save()
+            if not has_access(request.user, 'staff', course_key):
```

Comment:
@dianakhuang @cahrens let's check for all variations of course staff and discussion admin / moderator / TA. In most cases, someone who is a discussion admin is also course staff, but I'd rather be consistent and say that anyone who is staff at all on the course isn't automatically added, than have some staff added to a team upon creation and some not.